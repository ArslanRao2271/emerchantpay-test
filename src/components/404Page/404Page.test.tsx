import { render, screen } from "@testing-library/react";
import { PageNotFound } from ".";
import { MemoryRouter } from "react-router-dom";

const getWrapperComponent = (comp: JSX.Element) => {
  return render(comp, { wrapper: MemoryRouter });
};

describe("404 Page test cases", () => {
  test("Should match previous snapshot", () => {
    const { asFragment } = getWrapperComponent(<PageNotFound />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("Should render 404 page", () => {
    getWrapperComponent(<PageNotFound />);

    expect(screen.queryByText(/Page not found!/)).toBeVisible();
  });

  test("Should have home button", () => {
    getWrapperComponent(<PageNotFound />);

    expect(screen.queryByText(/Home/)).toBeVisible();
  });
});
