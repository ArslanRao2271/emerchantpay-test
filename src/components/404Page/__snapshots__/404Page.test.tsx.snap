// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 Page test cases Should match previous snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-7jugx5"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
    >
      Page not found!
    </h2>
    <a
      href="/"
    >
      Home
    </a>
  </div>
</DocumentFragment>
`;

exports[`Button test cases Should match previous snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-7jugx5"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
    >
      Page not found!
    </h2>
  </div>
</DocumentFragment>
`;
